components/nimbus/src/stateful/targeting.rs

```diff
@@ -27,3 +36,82 @@ impl NimbusTargetingHelper {
         }
     }
 }
+
+pub trait RecordedContext: Send + Sync {
+    /// Returns a JSON representation of the context object
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn to_json(&self) -> JsonObject;
+
+    /// Returns a HashMap representation of the event queries that will be used in the targeting
+    /// context
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn get_event_queries(&self) -> JsonObject;
+
+    /// Sets the object's internal value for the event query values
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn set_event_query_values(&self, json: JsonObject);
+
+    /// Records the context object to Glean
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn record(&self);
+}
+
+impl dyn RecordedContext {
+    pub fn execute_queries(
+        &self,
+        nimbus_targeting_helper: &NimbusTargetingHelper,
+    ) -> Result<Map<String, Value>> {
+        let results = Map::from_iter(self.get_event_queries().iter().filter_map(
+            |(key, value): (&String, &Value)| match value.as_str() {
+                Some(v) => match nimbus_targeting_helper.evaluate_jexl_raw_value(v.into()) {
+                    Ok(result) => Some((key.to_string(), result)),
+                    Err(err) => {
+                        log::info!(
+                            "error during jexl evaluation for query {} — {}",
+                            value,
+                            err.to_string()
+                        );
+                        None
+                    }
+                },
+                None => {
+                    log::info!("value {} for key {} is not a string", key, value);
+                    Some((key.to_string(), json!(value)))
+                }
+            },
+        ));
```

Comment:
Its more idiomatic to use Iterator::collect<T> to make a T when T: FromIterator. See [docs](https://doc.rust-lang.org/nightly/core/iter/trait.FromIterator.html)

We can also clean this up to use a single match:

```suggestion
    let results: Map<_, _> = m
        .iter()        
        .filter_map(|(key, value)|
            match value.as_str().map(|v| nimbus_targeting_helper.evaluate_jexl_raw_value(v)) {
                Some(Ok(v)) => Some((key.clone(), v)),
                Some(Err(e)) => {
                    log::info!(
                        "error during jexl evalaution for query {}={} - {}",
                        key,
                        value,
                        e,
                    );
                    None
                },
                None => {
                    log::info!("query {}={} does not map to string", key, value);
                    None
                },
            }      
        )
        .collect();

       
```